kernel/src/hil/symmetric_encryption.rs

```diff
@@ -1,39 +1,131 @@
-//! Interfaces for accessing encryption and decryption of symmetric ciphers.
+//! Interface for symmetric-cipher encryption
 //!
-//! Only AES-128-ctr supported at the moment.
-//!
-//! The interface is supposed to work for hardware supported crypto but should
-//! work for software implemented crypto as well.
-//!
-//! State Machine:
-//!
-//! 1. `init()`
-//! 2. `set_key()`
-//! 3. `aes128_crypt_ctr()`: can be used arbitrary number of times.
+//! see boards/imix/src/aes_test.rs for example usage
 
 use returncode::ReturnCode;
 
-pub trait SymmetricEncryption {
-    /// Set the callback client.
-    fn set_client(&self, client: &'static Client);
+/// Implement this trait and use `set_client()` in order to receive callbacks from an `AES128`
+/// instance.
+pub trait Client<'a> {
+    fn crypt_done(&'a self, source: Option<&'a mut [u8]>, dest: &'a mut [u8]);
+}
+
+/// The number of bytes used for AES block operations.  Keys and IVs must have this length,
+/// and encryption/decryption inputs must be have a multiple of this length.
+pub const AES128_BLOCK_SIZE: usize = 16;
```

Comment:
I noticed that despite commenting that you would add a constant for key size 16 days ago in response to @niklasad1 's suggestion, it appears you still assume all key sizes will be 128 bits. It doesn't look like this ended up happening, though I suppose that since all of the constant names imply this implementation is only for AES128 that is fine?